pkg/controller/association/dynamic_watches.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package association
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/certificates"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/watches"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+)
+
+// esWatchName returns the name of the watch setup on the referenced Elasticsearch resource.
+func esWatchName(associated types.NamespacedName) string {
+	return associated.Namespace + "-" + associated.Name + "-es-watch"
+}
+
+// esUserWatchName returns the name of the watch setup on the ES user secret.
+func esUserWatchName(associated types.NamespacedName) string {
+	return associated.Namespace + "-" + associated.Name + "-es-watch"
```

Comment:
"-user-watch"?

Reply:
😮  👍

Reply:
Not that it matters much, you don't have another watch with the same name on secrets iiuc